### PR TITLE
DM-1718, refactor file writing code. Part 2

### PR DESCRIPTION
**Pull request by SkyToGround**
### Issue

DM-1718

### Description of work

*Add a description of the work done. The aim is provide information to help the reviewer review the PR efficiently.
See [CONTRIBUTING.MD](CONTRIBUTING.MD) for more information on code standards etc.*

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: src/schemas/template/TemplateWriter.h [hunk 1]
```diff
@@ -221,7 +218,7 @@ class WriterClass : public FileWriter::HDFWriterModule {
   /// \param HDFAttributes Additional attributes as defined in the Nexus
   /// structure which the HDFWriterModule should write to the file. Because the
   /// HDFWriterModule is free to create the structure and datasets according to
-  /// its needs, it must also take the responsibility to write these
+  /// its needs, it must also take the reposnsibility to write these
```
Comment (mattclarke) [inline src/schemas/template/TemplateWriter.h]: Why did you add a typo?

Comment (mattclarke) [reply src/schemas/template/TemplateWriter.h]: ;)

Comment (SkyToGround) [reply src/schemas/template/TemplateWriter.h]: Good question. I do not know.

Comment (SkyToGround) [reply src/schemas/template/TemplateWriter.h]:
```suggestion
  /// its needs, it must also take the responsibility to write these
```

File: src/HDFWriterModule.h [hunk 2]
```diff
@@ -9,16 +9,16 @@
 
 #pragma once
 
+#include "FlatbufferMessage.h"
```
Comment (matthew-d-jones) [inline src/HDFWriterModule.h]: You don't like forward declarations? They can drastically decrease compile times...

File: src/Source.cpp [hunk 3]
```diff
@@ -8,6 +8,11 @@
 // Screaming Udder!                              https://esss.se
 
 #include "Source.h"
+#include "helper.h"
+#include "logger.h"
+#include <chrono>
+#include <fstream>
+#include <thread>
```
Comment (matthew-d-jones) [inline src/Source.cpp]: Why have all these includes appeared? It doesn't look like any code was added which need them.

File: src/Source.h [hunk 4]
```diff
@@ -12,7 +12,9 @@
 #include "FlatbufferReader.h"
 #include "HDFFile.h"
 #include "HDFWriterModule.h"
+#include "Msg.h"
 #include "ProcessMessageResult.h"
+#include "json.h"
```
Comment (matthew-d-jones) [inline src/Source.h]: Why the extra includes?

File: src/HDFWriterModule.h (hunk 2 above)
Comment (SkyToGround) [reply src/HDFWriterModule.h]: It's more about not actively searching for places where they can be used. I can add one in this case but if you know some tool that automates finding opportunities for forward declarations, that would be great.

Comment (matthew-d-jones) [reply src/HDFWriterModule.h]: I asked because it looks like you actively removed one here.
When writing new stuff I think it is just a case of being aware of the cost of adding includes to headers: forward declare by default and only `#include` if that doesn't work.

Comment (SkyToGround) [reply src/HDFWriterModule.h]: I can't be bother to figure out exactly why it happened but seeing as you added this forward declarations to this file very recently, it seems likely that the reason for it disappearing was because I branched off of master before that change had been merged.

Comment (SkyToGround) [reply src/HDFWriterModule.h]: Considering the changes I had to do to make the forward declaration work, my previous comment seems like a very likely reason.

File: src/Source.cpp (hunk 3 above)
Comment (SkyToGround) [reply src/Source.cpp]: I do not know but it is not entirely unlikely that the reason is something similar to the comment above this one. It has now been fixed.

File: src/Source.h (hunk 4 above)
Comment (SkyToGround) [reply src/Source.h]: Ditto.

**Comment by SkyToGround** [comment]
@matthew-d-jones Thanks for the review! I have fixed the issues that you pointed out.

**Comment by SkyToGround** [comment]
Looks like I have resolved the merge conflicts.
